pycaret/containers/models/time_series.py

```diff
@@ -562,26 +572,114 @@ def __init__(self, globals_dict: dict) -> None:
         )
 
 
-class RandomForestCdsDtContainer(TimeSeriesContainer):
+#################################
+#### REGRESSION BASED MODELS ####
+#################################
+
+# ===============#
+# LINEAR MODELS #
+# ===============#
+
+
+class LinearCdsDtContainer(TimeSeriesContainer):
     def __init__(self, globals_dict: dict) -> None:
         logger = get_logger()
         np.random.seed(globals_dict["seed"])
 
-        from sklearn.ensemble import RandomForestRegressor
+        from sklearn.linear_model import LinearRegression
 
         # TODO add GPU support
-
         gpu_imported = False
 
-        regressor_args = (
-            {
-                "random_state": globals_dict["seed"],
-                "n_jobs": globals_dict["n_jobs_param"],
-            }
-            if not gpu_imported
-            else {"seed": globals_dict["seed"]}
+        if globals_dict["gpu_param"] == "force":
+            from cuml.linear_model import LinearRegression
+
+            logger.info("Imported cuml.linear_model.LinearRegression")
+            gpu_imported = True
+        elif globals_dict["gpu_param"]:
+            try:
+                from cuml.linear_model import LinearRegression
+
+                logger.info("Imported cuml.linear_model.LinearRegression")
+                gpu_imported = True
+            except ImportError:
+                logger.warning("Couldn't import cuml.linear_model.LinearRegression")
+
+        regressor_args = {}
+        if not gpu_imported:
+            regressor_args["n_jobs"] = globals_dict["n_jobs_param"]
+        regressor = LinearRegression(**regressor_args)
+
+        args = {"regressor": regressor}
+        tune_args = {}
+        sp = globals_dict.get("seasonal_parameter")
+        sp = sp if sp is not None else 1
+        tune_grid = {
+            "sp": [sp],
+            "deseasonal_model": ["additive"],
+            "degree": [1],
+            "window_length": [10],
+            "regressor__fit_intercept": [True, False],
+            "regressor__normalize": [True, False],
+        }
+        tune_distributions = {
```

Comment:
Same here, it is not easy to follow the `__init__` method, maybe for all classes could go methods like `get_tune_grid`  and `get_tune_distributions`

Reply:
maybe these methods `get_tune_grid` and `get_tune_distributions` can be a part of the parent class `TimeSeriesExperiment`, we can actually force each container to have this method trough an `abstractmethod`.

Reply:
I like the idea. Currently, I was just following the approach that already existed in other modules, but this modularization is definitely something that I would like to have as well. Since it can potentially be leveraged by other tasks as well (regression, classification, etc.), lets quickly discuss with @Yard1 before making this change. I will open a Issue for this so we keep track of it,

Reply:
great, thanks Nikhil.